Feat/member read

yjn33 opened the pull request:
feat/member-read 본인 타인 유저프로필 기본 기능 구현 완료하였습니다.
본인 유저프로필 조회할때 현재 로그인한 http session에서 memberId를 받아와 findActiveMemberById
에 전달하는것으로 구현하였습니다. MemberController에서 @LoginRequired로 예외처리를 하기때문에
MemberService에 있는 findMyMember에서는 따로 예외처리를 안해도 될거같습니다.

yjn33 committed:
Feat(member-read): 멤버 프로필 조회 기능 구현

yjn33 committed:
Feat(member-read): 유저조회 추가

yjn33 committed:
Feat(member-read): 본인, 타인 유저 프로필 조회 기능 구현

MemberController.java -> findMemberById 구현 findMyMember 만드는중

MemberService.java -> findMemberById 구현 findMyMember 만드는중

yjn33 committed:
Fix(member-read): dev merge 충돌 수정

yjn33 committed:
Feat(member-read): 본인 유저프로필 조회 기본기능 구현

MemberController.java -> findMyMember메서드 현재 로그인한 http session을 서비스단에 전달하는것을 구현

MemberService.java -> findMyMember메서드 전달받은 http session에서 현재 로그인한 memberId를 추출하여 DB에서 유저 프로필을 조회

queenriwon commented on src/main/java/com/example/newsfeed/member/service/MemberService.java:
```diff
@@ -174,4 +179,68 @@ public Member loginMember(String email, String password) {
+
+            Member findMember = optionalMember.get();
+            return new FindMemberDto(findMember.getId(), findMember.getNickname(), findMember.getEmail(), findMember.getInfo(), findMember.getMbti(), findMember.getCreatedAt());
+        }
```
에러 발생 부분만 if조건문으로 잡고 else는 작성하지 않으셔도 같은 기능을 합니다.

queenriwon commented on src/main/java/com/example/newsfeed/member/service/MemberService.java:
```diff
@@ -174,4 +179,68 @@ public Member loginMember(String email, String password) {
+        // 현재 로그인 세션을 얻기
+        HttpSession session = httpServletRequest.getSession(false);
+        SessionMemberDto currMemberSession = (SessionMemberDto) session.getAttribute("member");
+
```
이부분 MemberController로 옮기고 SessionMemberDto를 매개변수로 가져오시면 controller와 service구분이 명확해질 것 같습니다.

queenriwon replied:
컨트롤러에서
@SessionAttribute(name = "member") SessionMemberDto session를 사용하시면 해당 코드는 사용하지 않으셔도 됩니다 :)

queenriwon commented on src/main/java/com/example/newsfeed/member/service/MemberService.java:
```diff
@@ -174,4 +179,68 @@ public Member loginMember(String email, String password) {
+             */
+            // status 404 Not Found 예외
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, memberId + "의 memberId값을 갖는 유저는 탈퇴했거나 존재하지 않습니다.");
+        }
```
Member member = memberRepository.findActiveMemberById(memberId)
    .orElseThrow( () -> new ResponseStatusException(...););

형식으로 작성하시면, 해당 부분 한줄로 작성하실 수 있습니다.
저 같은 경우는 이런 id를 통해 member를 찾는 코드는 많이 사용하기때문에, 메서드를 따로 빼서 사용하기도 합니다. 233번 줄에서도 비슷한 기능이 있으니 메서드를 하나 파셔서 재사용하는 방법도 고려해보셔용 (자신의 정보라서 예외처리를 하지않도록 하신 것 같은데, 방어로직 느낌으로 똑같이 적용이 되어도 된다고 생각합니다)

주석으로 repository에서 예외처리를 할 수 없다고 작성해주셨는데, 정확히는 Repository와 Service영역을 분리해주기 위해서 예외를 service에서 처리해주면 좋을 것 같다~~ 입니다.

queenriwon commented on src/main/java/com/example/newsfeed/member/controller/MemberController.java:
```diff
@@ -79,4 +80,34 @@ public ResponseEntity<Void> deleteMember(
+     */
+    @LoginRequired // 로그인을 안하면 LoginIntercepter.java 클래스를 통해 예외 발생 throw new RuntimeException("로그인 필요")
+    @GetMapping
+    public ResponseEntity<FindMyMemberDto> findMyMember(HttpServletRequest httpServletRequest)
```
HttpServletRequest httpServletRequest를 

@SessionAttribute(name = "member") SessionMemberDto session 로 바꾸면 getSession(), getAttribute()를 할 필요 없이 세션값을 바로 가져올 수 있습니다.

yjn33 replied:
올ㅋ 알겠습니다

queenriwon reviewed (commented):
코드작성 관련으로 조금 아쉽다 싶은 부분만 코멘트 남겨드렸습니다.
기능적으로는 문제없습니다^^ 수고하셨습니당

yjn33 committed:
Refactor(member-read): 로그인 세션 코드 리팩토링

findMemberById 메소드 -> 람다식으로 깔끔하게 정리

findMyMember 메소드 -> @SessionAttribute를 사용하여 현재 로그인세션 코드를 깔끔하게 정리

yjn33 committed:
Fix(member-read): dev merge 충돌 수정
